assets/js/Sign.jsx

```diff
@@ -13,40 +14,193 @@ function fontSize(signId) {
   return {};
 }
 
-function Sign({
-  signId,
-  lineOne,
-  lineOneDuration,
-  lineTwo,
-  lineTwoDuration,
-  currentTime,
-  line,
-  isEnabled,
-  setEnabled,
-  realtimeId,
-}) {
-  return (
-    <div className="viewer--sign">
-      <div className="viewer--sign-id" style={{ borderTopColor: lineToColor(line) }}>
-        <span style={fontSize(signId)}>
-          {signId}
-        </span>
-        <div>
-          <input id={realtimeId} type="checkbox" checked={isEnabled} className="ios8-switch" onChange={() => { setEnabled({ [realtimeId]: !isEnabled }); }} />
-          {/* eslint-disable-next-line jsx-a11y/label-has-for */}
-          <label htmlFor={realtimeId} />
-        </div>
-      </div>
-      <div className="viewer--sign-lines">
-        <div className="viewer--sign-line">
-          { displayLine(lineOneDuration, currentTime) ? lineOne : null }
-        </div>
-        <div className="viewer--sign-line">
-          { displayLine(lineTwoDuration, currentTime) ? lineTwo : null }
+function makeConfig(mode) {
+  if (mode === 'auto') {
+    return { mode: 'auto' };
+  }
+
+  if (mode === 'static_text') {
+    return {
+      mode: 'static_text', expires: null, line1: '', line2: '',
+    };
+  }
+
+  if (mode === 'headway') {
+    return {
+      mode: 'headway', expires: null,
+    };
+  }
+
+  return { mode: 'off', expires: null };
```

Comment:
Would this function maybe be a little bit more compact and readable if put together into one big if / else if / else thingamy?

Reply:
_I_ think so, but `eslint` complained at me. It wanted the function to have an unambiguous `return` at the bottom.